src/Microsoft.ML.AutoML/Utils/SplitUtil.cs

```diff
@@ -10,7 +10,48 @@ namespace Microsoft.ML.AutoML
 {
     internal static class SplitUtil
     {
-        public static (IDataView[] trainDatasets, IDataView[] validationDatasets) CrossValSplit(MLContext context, 
+        public static (IDataView[] trainDatasets, IDataView[] validationDatasets) CrossValSplit(MLContext context,
+            IDataView trainData, uint numFolds, string samplingKeyColumn, TaskKind taskKind, string labelColumnName)
+        {
+            (IEnumerable<IDataView> trainDatasets, IEnumerable<IDataView> validationDatasets) = CrossValSplit(context, 
+                trainData, numFolds, samplingKeyColumn);
+
+            if (taskKind != TaskKind.BinaryClassification)
+            {
+                return (trainDatasets.ToArray(), validationDatasets.ToArray());
+            }
+
+            // If we're running binary classification, discard splits where there is not at least one 
+            // true & one false label in the test set. Otherwise, scoring the dataset crashes because
+            // AUC cannot be computed.
+            var filteredTrainDatasets = new List<IDataView>();
+            var filteredValidationDatasets = new List<IDataView>();
+            for (var i = 0; i < trainDatasets.Count(); i++)
+            {
+                var validationDataset = validationDatasets.ElementAt(i);
+                var labelColumn = validationDataset.Schema.First(c => !c.IsHidden && c.Name == labelColumnName);
+                if (DatasetDimensionsUtil.ComputeCardinality<bool>(validationDataset, labelColumn, 2) < 2)
```

Comment:
This is an expensive solution for the uncommon issue; and all binary classification pipeline will entail the cost (normally) unnecessarily.  The specific issue being handled is binary classification learner need both classes in all folds. 

Recommend instead letting the cross-val fold fail, and catching the error; then removing the CV result. We need to check that some CV folds were successful.

Post checking is better as only the bad pipelines/splits entail the cost. The datasets most likely to hit the issue are small and fast. Otherwise, the large datasets (which are least likely to hit the issue) entail the most overhead.

Reply:
We default to cross-val mode only for small datasets. Additionally, the cardinality counting stops as soon as one value of each label is encountered. Combining these two facts, most of the time, I would imagine this operation would be fast. Although +1 -- worst-case run time (user manually specifies a high # of splits for a dataset that is sorted by label) would be 1 pass over each of the folds. Although, since each iteration requires several featurization & training passes over each fold, even in this edge case that a user likely wouldn't encounter, run time may not be too abysmal.

+1 that we should make cross val robust to failures on individual folds. However, we may in addition want to preserve these kinds of up-front checks. If we can inexpensively determine that a fold is invalid up front, why, in general, featurize & train that fold, and and have it fail every iteration? Also, as we're passing iteration exceptions back to the user, we may not want to pass back the semi-cryptic AUC exception (https://github.com/dotnet/machinelearning/issues/3800) back as a failure that occurred each iteration. In general, we may also not want to clog up our iteration failed exception telemetry unnecessarily